### ISSUE TITLE
write_dta creates large file size due to blank variables

### ISSUE THREAD

**Issue by MartinLBarron** [tidyverse/haven#558]
When writing a character variable to Stata where all the values are the same, write_dta translates the length of the variable differently depending upon if the variable contains blanks (ie. ""), contains a single space (" "), or contains NA.   If the variable contains a single space for all variables, it is converted into a string of length 2.  But if the variable is a string that is completely blank, it converts it to s string of length 2,045.  (If the variable is a character variable but contains all NAs, it is converted to a long. but that's not relevant here).

Storing blank variables as string2045 causes overly large files (since presumably a totally blank variable could be stored with a length of 1).  Could write_dta store out blank variables in a smaller Stata variable width?

Below is a reproducible example.  When the string variable is blank, the resulting Stata file is 19.5mb, when it has a single blank space, it is 50.6kb, when it is NA it is 60.4kb.

```r
a <- seq(1,1e4)
b <- as.character(rep(NA, 1e4))
c <- rep("",1e4)
d <- rep(" ",1e4)
df_asNA<- tibble(id=a, var=b)
df_asBlank<- tibble(id=a, var=c)
df_asSpace<- tibble(id=a, var=d)

write_dta(df_asNA, "asNA.dta") # 60.4kb
write_dta(df_asBlank, "asBlanks.dta") # 19.5mb
write_dta(df_asSpace, "asSpace.dta") # 50.6kb
```

**Comment by hadley** [comment]
Duplicate of #437, I think.

**Comment by gorcha** [comment]
Confirming that this has been fixed by #659, we now get:
```r
write_dta(df_asNA, "asNA.dta") # 60.4kb
write_dta(df_asBlank, "asBlanks.dta") # 50.6kb
write_dta(df_asSpace, "asSpace.dta") # 50.6kb
```
